src/test/java/org/opentripplanner/apis/gtfs/PatternByServiceDatesFilterTest.java

```diff
@@ -0,0 +1,200 @@
+import static org.opentripplanner.apis.gtfs.PatternByServiceDatesFilterTest.FilterExpectation.REMOVED;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.apis.gtfs.generated.GraphQLTypes.GraphQLServiceDateFilterInput;
+import org.opentripplanner.model.calendar.CalendarService;
+import org.opentripplanner.transit.model._data.TransitModelForTest;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.network.StopPattern;
+import org.opentripplanner.transit.model.network.TripPattern;
+import org.opentripplanner.transit.model.site.RegularStop;
+import org.opentripplanner.transit.model.timetable.ScheduledTripTimes;
+import org.opentripplanner.transit.model.timetable.Trip;
+import org.opentripplanner.transit.service.DefaultTransitService;
+import org.opentripplanner.transit.service.StopModel;
+import org.opentripplanner.transit.service.TransitModel;
+import org.opentripplanner.transit.service.TransitService;
+
+class PatternByServiceDatesFilterTest {
+
+  private static final TransitService EMPTY_SERVICE = new DefaultTransitService(new TransitModel());
+  private static final Route ROUTE_1 = TransitModelForTest.route("1").build();
+  private static final Trip TRIP = TransitModelForTest.trip("t1").withRoute(ROUTE_1).build();
+  private static final TransitModelForTest MODEL = new TransitModelForTest(StopModel.of());
+  private static final RegularStop STOP_1 = MODEL.stop("1").build();
+  private static final StopPattern STOP_PATTERN = TransitModelForTest.stopPattern(STOP_1, STOP_1);
+  private static final TripPattern PATTERN_1 = pattern();
+  private static final LocalDate DATE = LocalDate.parse("2024-05-27");
+
+  enum FilterExpectation {
+    REMOVED,
+    NOT_REMOVED,
+  }
+
+  private static TripPattern pattern() {
+    var pattern = TransitModelForTest
+      .tripPattern("1", ROUTE_1)
+      .withStopPattern(STOP_PATTERN)
+      .build();
+
+    var tt = ScheduledTripTimes
+      .of()
+      .withTrip(TRIP)
+      .withArrivalTimes("10:00 10:05")
+      .withDepartureTimes("10:00 10:05")
+      .build();
+    pattern.add(tt);
+    return pattern;
+  }
+
+  static List<Arguments> invalidRange() {
+    return List.of(
+      Arguments.of(parse("2024-05-02"), parse("2024-05-01")),
+      Arguments.of(parse("2024-05-03"), parse("2024-05-01"))
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("invalidRange")
+  void invalidRange(LocalDate start, LocalDate end) {
+    assertThrows(
+      IllegalArgumentException.class,
+      () -> new PatternByServiceDatesFilter(EMPTY_SERVICE, start, end)
+    );
+  }
+
+  static List<Arguments> validRange() {
```

Comment:
I'm not sure if it is confusing or not to have the same method name for the test and the arguments.

Reply:
I don't really have an opinion, so I changed it: 5d29a448b96537f3d74da4f67be57832b60737f2